### PR TITLE
remove redundant event for authorizations list;  update shared view model

### PR DESCRIPTION
**Pull request by ConstantinKV**
https://github.com/saltedge/sca-authenticator-android/issues/104

### CODE REVIEW

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt [hunk 1]
```diff
@@ -52,7 +52,10 @@ import com.saltedge.authenticator.sdk.tools.crypt.CryptoToolsAbs
 import com.saltedge.authenticator.sdk.tools.keystore.KeyStoreManagerAbs
 import com.saltedge.authenticator.tools.ResId
 import com.saltedge.authenticator.tools.postUnitEvent
-import kotlinx.coroutines.*
+import kotlinx.coroutines.CoroutineDispatcher
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt:55]: to much new imports in a file that didn't have any changes.

File: app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivity.kt [hunk 2]
```diff
@@ -107,17 +108,11 @@ class MainActivity : LockableActivity(), ViewModelContract, SnackbarAnchorContai
         viewModel.onRestartActivityEvent.observe(this, Observer { event ->
             event.getContentIfNotHandled()?.let { super.restartLockableActivity() }
         })
-        viewModel.onShowAuthorizationsListEvent.observe(this, Observer { event ->
```
Comment (mnewlive) [inline app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivity.kt:110]: why u rm it?

File: app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt (hunk 1 above)
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/features/authorizations/list/AuthorizationsListViewModel.kt:55]: Yours favorite `Optimize Imports` did it

File: app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivity.kt (hunk 2 above)
Comment (ConstantinKV) [reply app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivity.kt:110]: now you have two instances of AuthorizationsList in back stack. First instantiated by nav_graph, second by main activity new model.

**Pull request merged**